### PR TITLE
 Adding plugin version to logs

### PR DESCRIPTION
**Pull request by jathmel**
To make it easier on developer to address bug fixes made by suppliers Wayfair plugin version has been added to the PlentyMarkets logs generated by the Wayfair plugin.

### CODE REVIEW

File: src/Services/LoggingService.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 use Wayfair\Core\Contracts\LoggerContract;
 use Wayfair\Core\Helpers\AbstractConfigHelper;
+use Wayfair\Helpers\ConfigHelper;
```
Comment (jhoule86) [inline src/Services/LoggingService.php:12]: remove!

File: src/Services/LoggingService.php [hunk 2]
```diff
@@ -113,6 +124,7 @@ public function extractVars($loggingInfo): array {
     $method = $loggingInfo['method'] ?? null;
     $referenceType = $loggingInfo['referenceType'] ?? null;
     $referenceValue = (int) $loggingInfo['referenceValue'] ?? null;
+    $additionalInfo['wayfairPluginVersion'] = $this->version;
```
Comment (jhoule86) [inline src/Services/LoggingService.php:127]: (trivial) consider declaring `wayfairPluginVersion` in a constant.

File: src/Services/LoggingService.php [hunk 3]
```diff
@@ -18,8 +19,18 @@ class LoggingService implements LoggerContract {
...
   const ERROR = 'ERROR';
 
+  /**
+   * Stores the version of the plugin
+   * @var string $version
+   */
+  public $version;
+
+  /**
+   * Undocumented function
```
Comment (jhoule86) [inline src/Services/LoggingService.php:29]: doc block?

File: src/Core/Helpers/AbstractConfigHelper.php [hunk 4]
```diff
@@ -68,4 +68,9 @@ abstract class AbstractConfigHelper {
    */
   abstract public function hasBooted(): bool;
 
+  /**
+   * @return string
```
Comment (jhoule86) [inline src/Core/Helpers/AbstractConfigHelper.php:72]: more documentation on this.

File: src/Services/LoggingService.php [hunk 5]
```diff
@@ -121,7 +133,7 @@ public function extractVars($loggingInfo): array {
    * Checks if it is alright to log at a level lower than ERROR.
    * Logging at levels such as INFO and DEBUG prior to the end of the "boot" period
    * Causes issues.
-   * 
+   *
```
Comment (jhoule86) [inline src/Services/LoggingService.php:136]: doc blocks for the rest of the stuff in this contract, if you have the time.

File: src/Helpers/ConfigHelper.php [hunk 6]
```diff
@@ -113,7 +113,7 @@ public function getImportOrderSince() {
   /**
    * @return string
    */
-  public function getPluginVersion() {
+  public function getPluginVersion(): string {
```
Comment (jhoule86) [inline src/Helpers/ConfigHelper.php:116]: Double-check that the other use(s) of this method are okay with us adding the type restriction here.

File: src/Services/LoggingService.php [hunk 7]
```diff
@@ -113,6 +127,7 @@ public function extractVars($loggingInfo): array {
     $method = $loggingInfo['method'] ?? null;
     $referenceType = $loggingInfo['referenceType'] ?? null;
     $referenceValue = (int) $loggingInfo['referenceValue'] ?? null;
+    $additionalInfo[self::WAYFAIR_PLUGIN_VERSION] = $this->version;
```
Comment (mcloughlin77) [inline src/Services/LoggingService.php:130]: If $loggingInfo['additionalInfo'] was null on line 126 then this version information will not actually get set

Comment (jathmel) [reply src/Services/LoggingService.php:130]: I am not sure why it's not showing up but on my PR it gives an empty array if $loggingInfo['additionalInfo'] was falsy on line 126

Comment (mcloughlin77) [reply src/Services/LoggingService.php:130]: I should have looked at the next line as I was assuming the intent was also to set the loggingInfo if it wasn't set.

File: src/Core/Helpers/AbstractConfigHelper.php [hunk 8]
```diff
@@ -33,16 +33,22 @@ abstract class AbstractConfigHelper {
   const INVENTORY_ITEMS_PER_PAGE = 500;
 
   /**
+   * Retrieve the configuration client id for the global client
```
Comment (jhoule86) [inline src/Core/Helpers/AbstractConfigHelper.php:36]: configured client ID for connections to Wayfair's secure APIs.
there is (unfortunately) more than one client using it.

File: src/Core/Helpers/AbstractConfigHelper.php [hunk 9]
```diff
@@ -33,16 +33,22 @@ abstract class AbstractConfigHelper {
...
+   *
    * @return mixed
    */
   abstract public function getClientId();
 
   /**
+   * Retrieve the configuration client secret for the global client
```
Comment (jhoule86) [inline src/Core/Helpers/AbstractConfigHelper.php:43]: configured client secret for connections to Wayfair's secure APIs.
there is (unfortunately) more than one client using it.

File: src/Core/Helpers/AbstractConfigHelper.php [hunk 10]
```diff
@@ -33,16 +33,22 @@ abstract class AbstractConfigHelper {
...
    * @return mixed
    */
   abstract public function getClientSecret();
 
   /**
+   * Checks if an item already exists in cache and returns the key,
+   * if the item does not exists in cache adds item to cache
```
Comment (jhoule86) [inline src/Core/Helpers/AbstractConfigHelper.php:50]: - the cache etc are implementation details that should not be mentioned in the contract.
- what does "item" mean? this looks like it is getting a very specific value, called an Order Referrer Key.

File: src/Core/Helpers/AbstractConfigHelper.php [hunk 11]
```diff
@@ -53,6 +59,8 @@ abstract class AbstractConfigHelper {
   abstract public function getStockBufferValue();
 
   /**
+   * Retrieve the configuration values for setting up a test environment
```
Comment (jhoule86) [inline src/Core/Helpers/AbstractConfigHelper.php:62]: This checks if the plugin is configured to use the **test** / dryRun mode when communicating with the Wayfair APIs.
I don't know what you mean about test environments.

File: src/Services/LoggingService.php [hunk 12]
```diff
@@ -104,6 +116,8 @@ private function getPlentyMarketLoggerInstance(string $method, string $reference
   }
 
   /**
+   * Maps data from the PlentyMarket designed associative array to values of an array to be used by our different log methods
```
Comment (jhoule86) [inline src/Services/LoggingService.php:119]: For some reason, the previous team used "PlentyMarket" but we should always call it "PlentyMarkets" or "PM" for short.

Comment (jhoule86) [inline src/Services/LoggingService.php:119]: I think your description is backwards.
Doesn't this map the data from the associative array that Wayfair provides to the PlentyMarkets logging API's inputs?

File: src/Services/LoggingService.php [hunk 13]
```diff
@@ -121,7 +136,7 @@ public function extractVars($loggingInfo): array {
    * Checks if it is alright to log at a level lower than ERROR.
    * Logging at levels such as INFO and DEBUG prior to the end of the "boot" period
    * Causes issues.
```
[the remark this answers is not in the document]
Comment (jathmel) [reply src/Services/LoggingService.php:138]: I don't understand what you mean by the above!

[the remark this answers is not in the document]
Comment (mcloughlin77) [reply src/Services/LoggingService.php:138]: The need for checking whether we are currently currently booting when we log will no longer be needed once PM fixes the issue referenced in that link. Jeff probably wants you to add an addition to the comment above such as: "PlentyMarkets is currently addressing this issue and this check can be removed once they deploy the fix. See the related forum post for details: https://forum.plentymarkets.com/t/wayfair-log-levels-info-and-debug-not-working-for-loggable-module/581320/27"

[the remark this answers is not in the document]
Comment (jhoule86) [reply src/Services/LoggingService.php:138]: oh when github was being weird there was data loss. deleting.

File: src/Core/Helpers/AbstractConfigHelper.php [hunk 14]
```diff
@@ -33,16 +33,22 @@ abstract class AbstractConfigHelper {
...
 
   /**
+   * Retrieve the configured client secret for connections to Wayfair's secure APIs
+   *
    * @return mixed
    */
   abstract public function getClientSecret();
 
   /**
+   * Returns an Order Referrer Key
```
Comment (jhoule86) [inline src/Core/Helpers/AbstractConfigHelper.php:50]: The method name contains "ReferrerValue" and seems to imply that there is only one.
Is it actually returning an arbitrary Key, and not a value?

Comment (jathmel) [reply src/Core/Helpers/AbstractConfigHelper.php:50]: yea that should say value not key

File: src/Core/Helpers/AbstractConfigHelper.php [hunk 15]
```diff
@@ -33,16 +33,22 @@ abstract class AbstractConfigHelper {
...
 
   /**
+   * Retrieve the configured client secret for connections to Wayfair's secure APIs
+   *
    * @return mixed
    */
   abstract public function getClientSecret();
 
   /**
+   * Returns an Order Referrer value
```
Comment (jhoule86) [inline src/Core/Helpers/AbstractConfigHelper.php:50]: The ID of the Referrer for Wayfair

**Pull request merged**